java/ql/src/experimental/Security/CWE/CWE-400/ThreadResourceAbuse.ql

```diff
@@ -0,0 +1,85 @@
+/**
+ * @name Uncontrolled thread resource consumption
```

Comment:
Lots of duplication here, can these be better shared?

Reply:
Factored out common code into the library